base/base/cgroupsv2.cpp

```diff
@@ -71,3 +71,27 @@ fs::path cgroupV2PathOfProcess()
     return {};
 #endif
 }
+
+std::optional<std::string> getCgroupsV2PathContainingFile(std::string_view file_name)
+{
+    if (!cgroupsV2Enabled())
+        return {};
+
+    if (!cgroupsV2MemoryControllerEnabled())
```

Comment:
This check technically makes `getCgroupsV2PathContainingFile` specific to the cgroups memory controller.

Suggest to remove l. 80-81 _and_ remove `cgroupsV2MemoryControllerEnabled` altogether. If the memory controller was disabled (which is unlikely in itself), then the few places that currently call `cgroupsV2MemoryControllerEnabled` will simply return `{}` if they find no file. The outcome is the same.